src/storage/heap_file.c

```diff
@@ -25065,4 +25090,48 @@ heap_update_and_log_header (THREAD_ENTRY * thread_p, const HFID * hfid, const PG
   return NO_ERROR;
 }
 
+void
+heap_log_postpone_heap_append (THREAD_ENTRY * thread_p, const HFID * hfid, const OID * class_oid,
+                               const std::vector <VPID> heap_pages_array)
+{
+  // This append needs to be run on postpone after the commit.
+  // First create the log data required.
+  size_t array_size = heap_pages_array.size ();
+  size_t log_data_size = sizeof (HFID) + sizeof (OID) + sizeof (size_t) + array_size * sizeof (VPID);
+  char *log_data_buffer = (char *) db_private_alloc (NULL, log_data_size + MAX_ALIGNMENT);
+  char *log_data = PTR_ALIGN (log_data_buffer, MAX_ALIGNMENT);
+  LOG_DATA_ADDR log_addr = LOG_DATA_ADDR_INITIALIZER;
+
+  // Now populate the log data needed.
+  size_t offset = 0;
+
+  // HFID
+  HFID_COPY ((HFID *) (log_data + offset), hfid);
+  offset += sizeof (HFID);
+
+  // class_oid
+  COPY_OID ((OID *) (log_data + offset), class_oid);
+  offset += sizeof (OID);
+
+  // array_size
+  memcpy ((log_data + offset), &array_size, sizeof (size_t));
+  offset += sizeof (size_t);
+
+  // The array of VPID.
+  for (size_t i = 0; i < array_size; i++)
+    {
+      VPID_COPY ((VPID *) (log_data + offset), &heap_pages_array[i]);
```

Comment:
VPID_COPY will leave 2 bytes of hole. valgrind complains uninitialized memory write.
It might be better to make it silent. 
please look at `OR_PUT_VPID_ALIGNED`.